tests/thread_float/tests/01-run.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021 Otto-von-Guericke-Universität Magdeburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect("THREADS CREATED")
+    child.expect("")
+    child.expect("THREAD t1 start")
+    child.expect("THREAD t2 start")
+    child.expect("THREAD t3 start")
+    for _ in range(10):
+        child.expect(r"t(\d): (\d\d\d).\d+")
+        first_thread = int(child.match.group(1))
+        assert 141 == int(child.match.group(2))
```

Comment:
I'd be happier, since this is a float test, if the digits after the period would be included into group 2 and this would be
```suggestion
        assert 141 == float(child.match.group(2))
```

and not a comparison for equality, but a check if the value is within a certain ε.